Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -230,7 +238,9 @@ namespace PhysX
         requestRegion.Clamp(heightfieldAabb);
 
         // if dirty region invalid, recreate the entire heightfield, otherwise request samples
-        bool shouldRecreateHeightfield = m_shapeConfig == nullptr;
+        bool shouldRecreateHeightfield = (m_shapeConfig == nullptr)
+            || ((changeMask & Physics::HeightfieldProviderNotifications::HeightfieldChangeMask::CreateEnd)
+                == Physics::HeightfieldProviderNotifications::HeightfieldChangeMask::CreateEnd);
```

Comment:
Can't the comparison use `==` directly like at the beginning of the function?

 `changeMask == Physics::HeightfieldProviderNotifications::HeightfieldChangeMask::CreateEnd`

if `changeMask` is suppose to be a bit mask with multiple options set, then the checks at the beginning of this function need to be modified to be like this line instead.

Reply:
changeMask should indeed be a bit mask, so the first checks should be changed. This check has been changed to be correct.